der/src/header.rs

```diff
@@ -13,6 +13,8 @@ pub struct Header {
 
     /// Length of the encoded value
     pub length: Length,
+
+    constructed: bool,
```

Comment:
Hmm, not sure how I feel about the mixed visibility here. Maybe make it `pub` for now and perhaps we could have a followup to make all of the fields private? (I imagine that would be a bit more invasive)

Reply:
Yep, definitely ugly.

Revised in https://github.com/RustCrypto/formats/pull/1931/commits/544e39783df87e5353a3691a37eb83928d87e400

---

> and perhaps we could have a followup to make all of the fields private? (I imagine that would be a bit more invasive)

I counted 24 occurrences in the der crate; for now I also access accessor methods so that new code has the option to use them in the event that the fields become private.

Reply:
I went through with it: https://github.com/RustCrypto/formats/pull/1931/commits/0c5789e98ae2f42cdd07cb7378bc23c7b464d738

However, tests are failing due to at least
https://github.com/RustCrypto/signatures/blob/1ff759e67a1675d759e901fdb5903a07a1d7caec/ecdsa/src/der.rs#L206-L210

It looks like this change would have to be phased in?

Should I
1. roll back all the changes for using accessor methods rather than directly accessing struct fields, OR
2. roll back making the fields private but keep the pattern of using accessor methods rather than direct field access
?

Reply:
Looks great so far! Easier than I expected, actually

You can open a PR to https://github.com/RustCrypto/signatures which makes the necessary changes, using a `[patch.crates-io]` directive to pull in this `git` branch.

After that, in a bit of an ouroboros you can then update this PR with a `[patch.crates-io]` directive to pull in `ecdsa` from your branch on https://github.com/RustCrypto/signatures. That should get the tests passing.

Reply:
Ok, I think I followed your guidance

1. https://github.com/RustCrypto/signatures/compare/master...dwhjames:RustCrypto-signatures:tmp_der_header_refactor
2. https://github.com/RustCrypto/formats/pull/1931/commits/fa5700cd9ce69d52af8c2ce0d86a06a579cb4ea7